scripts/devel/build_and_test.py

```diff
@@ -67,13 +67,14 @@ def main(argv=sys.argv[1:]):
         with Scope('SUBSECTION', 'build workspace in isolation'):
             test_results_dir = os.path.join(
                 args.workspace_root, 'test_results')
-            cmake_args = [
-                '-DBUILD_TESTING=1',
-                '-DCATKIN_ENABLE_TESTING=1', '-DCATKIN_SKIP_TESTING=0',
```

Comment:
Has the diff on this file been deployed to standard devel jobs to check for regressions?

Was there a problem when specifying these arguments with colcon (for ROS 2)?

I'm slightly concerned about equating colcon use with ROS 2 even if that's the likely case today.

Reply:
I believe the motivation is to reduce the number of CMake warnings because of unused command line definitions.

I don't think this is segregating the variables by ROS version - it just skips the `CATKIN_*` variables when we're using colcon.

Reply:
These arguments are still needed when building ROS 1 / catkin packages so they can't be removed. They are not specific to the build tool but to the build system used by the processed packages.